### PR TITLE
Provide options to parse Protobuf from classpath

### PR DESCRIPTION
**Pull request by devinrsmith**
Additionally, simplify the implementation to use the compatibility guarantees provided by protobuf instead of manually handling schema-change logic.

### CODE REVIEW

File: py/server/deephaven/stream/kafka/consumer.py [hunk 1]
```diff
@@ -275,19 +278,44 @@ def _consume(
     except Exception as e:
         raise DHError(e, "failed to consume a Kafka stream.") from e
 
+class ProtobufProtocol(JObjectWrapper):
+    """The protobuf serialization / deserialization protocol."""
+
+    j_object_type = jpy.get_type("io.deephaven.kafka.protobuf.Protocol")
+
+    @staticmethod
+    def serdes():
```
Comment (chipkent) [inline py/server/deephaven/stream/kafka/consumer.py:287]:
```suggestion
    def serdes() -> ProtobufProtocol:
```

File: py/server/deephaven/stream/kafka/consumer.py [hunk 2]
```diff
@@ -275,19 +278,44 @@ def _consume(
...
+        """The Kafka Protobuf serdes protocol. The payload's first byte is the serdes magic byte, the next 4-bytes are
+        the schema ID, the next variable-sized bytes are the message indexes, followed by the normal binary encoding of
+        the Protobuf data."""
+        return ProtobufProtocol(ProtobufProtocol.j_object_type.serdes())
+
+    @staticmethod
+    def raw():
```
Comment (chipkent) [inline py/server/deephaven/stream/kafka/consumer.py:294]:
```suggestion
    def raw() -> ProtobufProtocol:
```

File: py/server/deephaven/stream/kafka/consumer.py [hunk 3]
```diff
@@ -275,19 +278,44 @@ def _consume(
...
         schema_version: Optional[int] = None,
         schema_message_name: Optional[str] = None,
+        message_class: Optional[str] = None,
         include: Optional[List[str]] = None,
+        protocol: ProtobufProtocol = ProtobufProtocol.serdes(),
 ) -> KeyValueSpec:
-    """Creates a spec for how to use Kafka data when consuming a Kafka stream to a Deephaven table. This will fetch the
-    protobuf descriptor for the schema subject from the schema registry using version schema_version and create protobuf
-    message parsing functions according to parsing options. These functions will be adapted to handle schema changes.
+    """Creates a spec parsing a Kafka protobuf stream into a Deephaven table.
```
Comment (chipkent) [inline py/server/deephaven/stream/kafka/consumer.py:314]:
```suggestion
    """Creates a spec for parsing a Kafka protobuf stream into a Deephaven table.
```

**Review by chipkent — COMMENTED** [review]
My review is just of python.

File: py/server/deephaven/stream/kafka/consumer.py (hunk 1 above)
Comment (devinrsmith) [reply py/server/deephaven/stream/kafka/consumer.py:287]: Unfortunately, Python throws an error when trying to reference this class.

File: py/server/deephaven/stream/kafka/consumer.py (hunk 2 above)
Comment (devinrsmith) [reply py/server/deephaven/stream/kafka/consumer.py:294]: Unfortunately, Python throws an error when trying to reference this class.

Comment (chipkent) [reply py/server/deephaven/stream/kafka/consumer.py:294]: Try `def raw() -> 'ProtobufProtocol':`

File: py/server/deephaven/stream/kafka/consumer.py (hunk 1 above)
Comment (chipkent) [reply py/server/deephaven/stream/kafka/consumer.py:287]: Try `def serdes() -> 'ProtobufProtocol':`

**Review by chipkent — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

How-to: https://github.com/deephaven/deephaven.io/issues/3281
Reference: https://github.com/deephaven/deephaven.io/issues/3280

**Pull request merged**